accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,198 @@
+- Feature Name: payg_cloud
+- Start Date: 2021-08-11
+
+# Summary
+[summary]: #summary
+
+In the three major Public Clouds (AWS, GCP and Azure), SUSE:
+ - provides customized pay-as-you-go-specific product images (eg. SLES, SLES for SAP...)
+ - operates per-region RMT Servers mirroring repositories for products available as PAYG
+
+PAYG instances all come with Cloud-specific "tokens" that authorizes the specific instance to download specific products from SUSE-operated RMT Servers. PAYG instances are registered to the closest RMT Server at launch time (region/Server name is auto-determined).
+
+At update time, a zypper plugin forwards such "token" to the RMT Server via custom headers.
+To check repository accessibility RMT Servers have plugins which use Cloud-specific internal APIs to check whether the "token" is valid and which products it is entitled to.
+
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni needs to load repository data from external sources to provide it to the registered machines.
+At the moment Uyuni can syncronize repositores from:
+  - the SCC CDN
+  - a plain RMT Server ("from-mirror setup", still requires connection to SCC)
+  - a plain directory exported from an RMT Server ("disconnected setup")
+  - a custom repository
+
+Presently, it is not possible to sync content from a SUSE-operated Cloud RMT Server, because:
+  - a Uyuni Server does not come with "tokens" (unlike PAYG instances, see above)
+  - Uyuni does not know how to pass them to RMT Servers
+  - we do not know which repositories are served via such RMT Servers
+
+Since Uyuni cannot contact the cloud RMT servers the only option for users is to contact SUSE and get one SCC account which allows to synchronize content from the SCC CDN.
+The process for this is not straightforward and adds extract complexity to users and SUSE teams.
+
+The goal for this RFC is to propose a solution to simplify this process and allowed Uyuni to synchronize content directly from RMT cloud servers in a user friendly way.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Our solution will be based on the following steps:
+  - Connect to PAYG instance, extract authentication data to uyuni server
+  - Teach Uyuni how to use PAYG authentication data to reposync product repositories
+
+With this solution the expected user flow would be:
+  - Provide to Uyuni ssh information to connect to the PAYG instance
+    - uyuni will extract data from the PAYG instance
+    - product of the PAYG instance will then be displayed on the Products Setup Wizard
+  - Import product from the products setup wizard
```

Comment:
I'm not sure about this, since we are reusing the existing mechanism to add channels for products. Implement it via API to this PAYG solution will also allow to add product through API

Reply:
Hmm, maybe we are actually talking about the same thing.

What I mean is that the Uyuni Server has a feature to "add products", which can be used either via the UI, and in that case it's called "the Setup Wizard pages", or via API, and in that case it's called "sync.content" - and now that I think about it, we also have a third way which is via the CLI, `mgr-sync`.

The way the current text reads it sounds like we are looking for a UI-only solution, while actually we aim for a "core" solution that will extend to the API as well - actually possibly API-first, IIUC.

How does that sound?

Reply:
Makes totally sense. I have updated the text for clarity. Thanks